src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -51,6 +56,45 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
         
     }
 
+    @Override
+    public ChallengeDTO updateChallenge(Long challengeId, ChallengeDTO challengeDTO) throws NotFoundException {
+
+        Long userId;
+
+        if(challengeDTO.getChallengeCategory() != null) {
+
+            Optional<Long> challengeCategoryOptional = Optional.ofNullable(challengeDTO.getChallengeCategory().getId());
+
+            if (!challengeCategoryOptional.isPresent()) {
+                challengeDTO.setChallengeCategory(null);
+            }
+
+            challengeCategoryOptional.ifPresent(id -> challengeCategoryRepository.findById(id).
+                    orElseThrow(() -> new InvalidFieldException().setMessage("Categoria aleasa trebuie sa existe"))
+            );
+        }
+
+        if(!challengeId.equals(challengeDTO.getId()))
+        {
+            throw new InvalidFieldException().setMessage("Id-ul challenge-ului din path trebuie ca corespunda cu cel din DTO");
+        }
+
+        Challenge oldChallenge = challengeRepository.findById(challengeId).orElseThrow(
+                ()->new NotFoundException().setMessage("Challenge-ul " + challengeId +" nu exista")
+        );
+
+        userId = challengeDTO.getCreator().getId();
+
+        if(!oldChallenge.getCreator().getId().equals(userId))
+        {
+            throw new InvalidFieldException().setMessage("Creatorul unui challenge nu se poate schimba");
+        }
+
+        Challenge updatedChallenge = challengeRepository.save(ChallengeMapper.convertChallengeDTOToChallenge(challengeDTO));
+
+        return ChallengeMapper.convertChallengeToChallengeDTO(updatedChallenge);
```

Comment:
La linia 93 faci conversie de la Dto la entity, iar la linia 95 faci conversia inversa. De ce? nu poti returna acelas dto pe care l-ai convertit in entitate?